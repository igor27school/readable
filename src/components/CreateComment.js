import React, { Component } from 'react'
import { connect } from 'react-redux'
import serializeForm from 'form-serialize'
import { Link } from 'react-router-dom'
import { createComment } from '../actions/ActionCreators'

class CreateComment extends Component {
  handleSubmit = (e) => {
    e.preventDefault()
    const { parentId } = this.props.match.params
    const values = serializeForm(e.target, { hash: true })
    this.props.createComment(parentId, values)
    this.props.history.push(`/posts/${parentId}`)
  }
  render() {
    const { parentId } = this.props.match.params
    return (
      <div>
        <Link to={`/posts/${parentId}`}>Close</Link>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="author" placeholder="Author"/>
          <textarea name="body" placeholder="Body"></textarea>
          <button>Publish</button>
        </form>
      </div>
    )
  }
}

function mapDispatchToProps (dispatch) {
  return {
    createComment: (parentId, values) => dispatch(createComment(parentId, values)),
  }
}

export default connect(null, mapDispatchToProps)(CreateComment)
