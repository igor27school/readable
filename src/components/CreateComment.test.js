import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { CreateComment } from './CreateComment'

jest.mock('../utils/ServerAPI')

describe('CreateComment tests', () => {
  const setup = () => {
    const props = {
      createComment: jest.fn(),
      match: {
        params: {
          category: 'testCategory',
          parent_id: 'testPostId',
        }
      }
    }
    const renderer = new ShallowRenderer()
    return {
      props,
      renderer
    }
  }

  it('CreateComment matches the snapshot', () => {
    const { renderer, props } = setup()
    renderer.render(<CreateComment {...props}/>)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
