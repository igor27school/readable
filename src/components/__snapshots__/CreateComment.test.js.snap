// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateComment tests CreateComment matches the snapshot 1`] = `
<div>
  <Link
    replace={false}
    to="/testCategory/testPostId"
  >
    Close
  </Link>
  <form
    onSubmit={[Function]}
  >
    <input
      name="author"
      placeholder="Author"
      type="text"
    />
    <textarea
      name="body"
      placeholder="Body"
    />
    <button>
      Publish
    </button>
  </form>
</div>
`;
